pkgs/tools/networking/pihole/default.nix

```diff
@@ -0,0 +1,106 @@
+{ stdenv
+, lib
+, fetchFromGitHub
+, makeWrapper
+, installShellFiles
+, coreutils
+, curl
+, dig
+, gawk
+, glibc
+, gnugrep
+, gnused
+, iproute2
+, killall
+, netcat
+, pihole-ftl
+, sqlite
+, systemd
+, stateDir ? "/etc/pihole"
+, ...
+}:
+
+stdenv.mkDerivation rec {
+  pname = "pihole";
+  version = "5.16.2";
+
+  src = fetchFromGitHub {
+    owner = "pi-hole";
+    repo = "pi-hole";
+    rev = "v${version}";
+    sha256 = "sha256-9X9FgctfE8odp6b6SOT5nT6jqjW6zuwKbBNt40zicMk=";
+  };
+
+  patches = [
+    ./0001-NixOS-support.patch
+    ./0002-Patch-and-remove-unsupported-commands.patch
+    ./0003-Add-DONT_RESTART_FTL-option.patch
+  ];
+
+  nativeBuildInputs = [ makeWrapper installShellFiles ];
+
+  installPhase =
+    let
+      inputs = [
+        coreutils
+        curl
+        dig
+        gawk
+        glibc
+        gnugrep
+        gnused
+        iproute2
+        killall
+        netcat
+        pihole-ftl
+        sqlite
+        systemd
+      ];
+    in
+    ''
+      runHook preInstall
+
+      readonly scriptsDir=$out/usr/share/pihole
+      mkdir -p $scriptsDir
+      cp --parents -r -t $scriptsDir/ advanced/{Scripts,Templates}/
+
+      mkdir -p $out/bin
+
+      cp -t $out/bin pihole
+
+      cp -t $scriptsDir/advanced/Scripts gravity.sh
+      chmod +x $scriptsDir/advanced/Scripts/gravity.sh
+
+      substituteInPlace $out/bin/pihole advanced/Scripts/pihole-reenable.sh \
+        --replace '"''${PI_HOLE_BIN_DIR}"/pihole' "$out/bin/pihole"
+
+      substituteInPlace $out/bin/* $scriptsDir/advanced/Scripts/*.sh \
+        --replace /opt/pihole $scriptsDir/advanced/Scripts \
+        --replace /etc/.pihole $scriptsDir
+
+      substituteInPlace $scriptsDir/advanced/Scripts/gravity.sh \
+        --replace 'PIHOLE_COMMAND="/usr/local/bin/''\${basename}"' PIHOLE_COMMAND=$out/bin/pihole
+
+      wrapProgram $out/bin/pihole \
+        --prefix PATH : ${lib.makeBinPath inputs}
+
+      wrapProgram $scriptsDir/advanced/Scripts/gravity.sh \
+        --prefix PATH : ${lib.makeBinPath inputs}
```

Comment:
Why not use resholve?

Reply:
Didn't know it existed, I'll take a look